apps/src/code-studio/progressRedux.ts

```diff
@@ -0,0 +1,493 @@
+import $ from 'jquery';
+import _ from 'lodash';
+import {
+  Lesson,
+  LessonGroup,
+  UnitProgress,
+  PeerReviewLessonInfo,
+  PeerReviewSummary,
+  PUZZLE_PAGE_NONE,
+  InitProgressPayload,
+  LevelResults,
+  ViewType,
+  PeerReviewLevelInfo,
+} from '@cdo/apps/types/progressTypes';
+import {
+  AnyAction,
+  PayloadAction,
+  ThunkAction,
+  ThunkDispatch,
+  createSlice,
+} from '@reduxjs/toolkit';
+import {
+  processServerStudentProgress,
+  getLevelResult,
+} from '@cdo/apps/templates/progress/progressHelpers';
+import {mergeActivityResult} from './activityUtils';
+import {SET_VIEW_TYPE} from './viewAsRedux';
+import {setVerified} from '@cdo/apps/code-studio/verifiedInstructorRedux';
+import {authorizeLockable} from './lessonLockRedux';
+import {updateBrowserForLevelNavigation} from './browserNavigation';
+import {TestResults} from '@cdo/apps/constants';
+
+interface ProgressState {
+  currentLevelId: string | null;
+  currentLessonId: number | undefined;
+  deeperLearningCourse: boolean | null;
+  saveAnswersBeforeNavigation: boolean | null;
+  lessons: Lesson[] | null;
+  lessonGroups: LessonGroup[] | null;
+  scriptId: number | null;
+  scriptName: string | null;
+  scriptDisplayName: string | undefined;
+  unitTitle: string | null;
+  courseId: number | null;
+  isLessonExtras: boolean;
+  unitProgress: {
+    [key: number]: UnitProgress;
+  };
+  unitProgressHasLoaded: boolean;
+  levelResults: LevelResults;
+  focusAreaLessonIds: number[];
+  peerReviewLessonInfo: PeerReviewLessonInfo | null;
+  peerReviewsPerformed: PeerReviewSummary[];
+  postMilestoneDisabled: boolean;
+  isAge13Required: boolean;
+  studentDefaultsSummaryView: boolean;
+  isSummaryView: boolean;
+  isMiniView: boolean;
+  hasFullProgress: boolean;
+  lessonExtrasEnabled: boolean;
+  usingDbProgress: boolean;
+  currentPageNumber: number;
+  courseVersionId: number | undefined;
+  unitDescription: string | undefined;
+  unitStudentDescription: string | undefined;
+  changeFocusAreaPath: string | undefined;
+  unitCompleted: boolean | undefined;
+}
+
+const initialState: ProgressState = {
+  currentLevelId: null,
+
+  // These first fields never change after initialization
+  currentLessonId: undefined,
+  deeperLearningCourse: null,
+  // used on multi-page assessments
+  saveAnswersBeforeNavigation: null,
+  lessons: null,
+  lessonGroups: null,
+  scriptId: null,
+  scriptName: null,
+  scriptDisplayName: undefined,
+  unitTitle: null,
+  courseId: null,
+  isLessonExtras: false,
+
+  // The remaining fields do change after initialization
```

Comment:
Maybe worth separating this important line?

Reply:
Do you mean add a blank line after it? I'll do that

Reply:
Yes, and maybe a period at the end for good measure :)